examples/dmd/de_dg_advection_greedy.cpp

```diff
@@ -0,0 +1,1635 @@
+            fout << to_string(f_factor) << " : " << tot_diff_norm_u /
+                 tot_true_solution_u_norm << std::endl;
+            fout.close();
+        }
+
+        delete result_U;
+    }
+    // 22. Calculate the relative error as commanded by the greedy algorithm.
+    if (offline)
+    {
+        if (myid == 0) std::cout << "The relative error is: " << rel_diff <<
+                                     std::endl;
+        greedy_sampler->setPointRelativeError(rel_diff);
+    }
+    // 23. Or calculate the error indicator as commanded by the greedy algorithm.
+    else if (calc_err_indicator)
+    {
+        if (myid == 0) std::cout << "The error indicator is: " << rel_diff << std::endl;
+        greedy_sampler->setPointErrorIndicator(rel_diff, 1);
+    }
+
+    if (!online && myid == 0)
+    {
+        printf("Elapsed time for solving FOM: %e second\n", fom_timer.RealTime());
+    }
+
+    // 24. Free the used memory.
+    delete ode_solver;
+    delete pmesh;
+    if (dmd_U != NULL)
+    {
+        delete dmd_U;
+    }
+
+    return rel_diff;
+}
+
+class RelativeDifferenceCostFunction : public CAROM::IOptimizable
+{
+public:
+    RelativeDifferenceCostFunction(unsigned int dims) :
+        m_dim(dims)
+    {
+
+    }
+    double EvaluateCost(std::vector<double> inputs) const override
+    {
+        f_factor = inputs[0];
+
+        return simulation();
+    }
+
+    unsigned int NumberOfParameters() const override
+    {
+        return m_dim;
+    }
+
+    std::vector<Constraints> GetConstraints() const override
+    {
+        std::fstream fin("parameters.txt", std::ios_base::in);
+        double curr_param;
+        bool first_line = true;
+        double min_f_factor = 0.0;
+        double max_f_factor = 0.0;
+
+
+        while (fin >> curr_param)
+        {
+            double curr_f_factor = curr_param;
+
+            if (first_line)
+            {
+                min_f_factor = curr_f_factor;
+                max_f_factor = curr_f_factor;
+                first_line = false;
+            }
+            else
+            {
+                min_f_factor = min(curr_f_factor, min_f_factor);
+                max_f_factor = max(curr_f_factor, max_f_factor);
+            }
+        }
+        fin.close();
+
+        if (de_min_f_factor != -DBL_MAX) min_f_factor = de_min_f_factor;
+        if (de_max_f_factor != DBL_MAX) max_f_factor = de_max_f_factor;
+
+        MFEM_VERIFY(min_f_factor <= max_f_factor, "Radius DE range is invalid.");
+
+        if (myid == 0) cout << "DE f_factor range is: " << min_f_factor << " to " <<
+                                max_f_factor << endl;
+
+        std::vector<Constraints> constr(NumberOfParameters());
+        constr[0] = Constraints(min_f_factor, max_f_factor, true);
+        return constr;
+    }
+
+private:
+    unsigned int m_dim;
+};
+
+int main(int argc, char *argv[])
+{
+
+    cout.precision(precision);
+    // 2. Parse command-line options.
+// MFEM parameters.
```

Comment:
The style check passes, so this is officially ok, but we usually indent comments rather than starting at the first column.

Reply:
Indented comments and reran the style check.